### DISCUSSION TITLE
T. in empty cell test should not spawn threads.

### DISCUSSION THREAD

**Discussion by rdm** [jsoftware/jsource#261] [Ideas]
See thread leading to http://www.jsoftware.com/pipermail/beta/2022-April/010458.html for details

**Comment by moon-chilled** [comment]
I suggest that a general mechanism be devised, by which a user verb is also able to detect such a situation and perhaps skip side effects.

**Comment by rdm** [comment]
Something like returnifempty. keyword? Or maybe dolittle. would be more poetic...

That said, adding to the language's vocabulary is out of scope here. This particular issue is about addressing an issue with the threading vocabulary.

A proposal for new vocabulary should belong in a different issue, where it could be discussed and treated adequately.

**Comment by HenryHRich** [comment]
The issue is execution on fill-cells, nothing specific to T. .  You can use ^:(*@#@]) to suppress execution on empty, or use u&.> which does not bother with execution on a fill-cell
